blog/why-we-chose-tidb-to-support-ossinsight-v2/index.md

```diff
@@ -0,0 +1,236 @@
+
+**Thanks to GH Archive, we found the data source.**
+
+But there's another problem: hourly data is good, but not good enough. We wanted our data to be updated in real time—or at least near real time. We decided to directly use the [GitHub event API](https://docs.github.com/en/rest/activity/events), which collects all events that have occurred within the past hour. 
+
+By combining the data from the GH Archive and the GitHub event API, we can gain streaming, real-time event updates.
+
+
+<br/>
+
+![GitHub event updates](./github-events-updates.gif)
+
+<center><em>GitHub event updates</em></center>
+
+<br/>
+
+
+## Headache 2: the data is huge!
+
+After we decompressed all the data from GH Archive, we found there were more than 4.6 billion rows of GitHub events. That’s a lot of data!  We also noticed that about 300,000 rows were generated and updated each hour.
+
+<br/>
+
+![The data volume of GitHub events occurred after 2011](./github-events.png)
+
+<center><em>The data volume of GitHub events occurred after 2011</em></center>
+
+<br/>
+
+The database solution would be tricky here. Our goal is to build an application that provides real-time data insights based on a continuously growing dataset. So, scalability is a must. NoSQL databases can provide good scalability, but what follows is how to handle complex analytical queries. Unfortunately, NoSQL databases are not good at that. 
+
+[image to be updated]
+
+Another option is to use an OLAP database such as ClickHouse. ClickHouse can handle the analytical workload very well, but it is not designed for serving online traffic. If we chose it, we would need another database for the online traffic. 
+
+[image to be updated]
+
+What about sharding the database and then building an extract, transform, load (ETL) pipeline to synchronize the new events to a data warehouse? This sounds workable.
+
+<br/>
+
+![How a sharded database solution can handle the GitHub data](./sharded-architecture.png)
+
+<center><em>How a sharded database solution can handle the GitHub data</em></center>
+
+<br/>
+
+According to our project manager's (PM’s) plan, we needed to do some repo-specific or user-specific analysis. Although the total data volume was huge, the number of events was not too large for a single project or user. This meant using the secondary indexes in RDBMS would be a good idea. But, if we decided to use the above architecture, we had to be careful in selecting the database sharding key. For example, if we use `user_id` as the sharding key, then queries based on `repo_id` will be very tricky. 
+
+Also, we're not experts on Kafka and data warehouses, mastering and building such an infrastructure in just one week was a very difficult task for us.
+
+The choice is obvious now, and don't forget PingCAP is a database company! TiDB seems a perfect fit for this, and it's a good chance to eat our own dog food. So, why not using TiDB! :) 
```

Comment:
Nice chapter! Don't feel bad about the relationship between PingCAP and OSSinsight, you're working on a great project!

ping @TennyZhuang @tisonkun, Does this chapter looks good to you?